runtime/compiler/trj9/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -11940,32 +11990,51 @@ inlineStringHashCode(
       TR::Register *xmm2 = cg->allocateRegister(TR_FPR);
 
       generateLabelInstruction(LABEL, node, startLabel, cg);
+      // endReg = countReg + indexReg
       generateRegRegInstruction(MOVRegReg(), node, endReg, countReg, cg);
       generateRegRegInstruction(ADDRegReg(), node, endReg, indexReg, cg);
+      // zero out hashReg with xor
       generateRegRegInstruction(XORRegReg(), node, hashReg, hashReg, cg);
       generateRegImmInstruction(CMPRegImm4(), node, countReg, 0x4, cg);
       cg->stopUsingRegister(countReg);
       generateLabelInstruction(JL4, node, SerialLabel, cg);
 
       // SSE version
       // xmm2 = load 4 chars and expand into 4 int
+      // decrease endReg by 3 so that the SSE loop will always operate on 4 available characters
       generateRegImmInstruction(SUBRegImms(), node, endReg, 3, cg);
+      // 31^4 = 923521
       int vector1[4] = {923521, 923521, 923521, 923521};
       TR::IA32ConstantDataSnippet *vector1Snippet   = cg->findOrCreate16ByteConstant(node, vector1);
       TR::MemoryReference       *vector1MR = generateX86MemoryReference(vector1Snippet, cg);
       generateRegMemInstruction(MOVAPSRegMem, node, xmm0, vector1MR, cg);
       generateRegRegInstruction(XORPSRegReg, node, xmm1, xmm1, cg);
 
       generateLabelInstruction(LABEL, node, SSELabel, cg);
-      generateRegMemInstruction(MOVSDRegMem,
-                                node,
-                                xmm2,
-                                generateX86MemoryReference(valueReg, indexReg, 1, TR::Compiler->om.contiguousArrayHeaderSizeInBytes(), cg),
-                                cg);
+
+      // move 8 bytes (4 2-byte characters) to xmm2. Stride shift = 1 (each element in valueReg is 2 bytes. Used with indexReg for correct traversal)
+      if(!isCompressed)
```

Comment:
This if/else has both paths. Would it be possible to turn this into a positive query and avoid the negation?

Reply:
@fjeremic consulting OMR's compiler/z/codegen's inlineStringHashCode as a reference, it had the same type of negation if/else structure so I used that format. However if you'd like I can change it to become a positive query.

https://github.com/eclipse/omr/blob/ec2c3bf81b14c6ee01f5e03e8a35a730682ebb6a/compiler/z/codegen/OMRTreeEvaluator.cpp#L17252

Reply:
I think avoiding the negation makes sense for new code - there are certainly dark areas in the existing code worthy of clean up but if we make things a quantum better every time a change is made the overall quality of everything moves up over time.

Reply:
sure, I'll make the change

Reply:
> However if you'd like I can change it to become a positive query.

Definitely fix up both while you're in this area of code. Thanks!